src/interpreter/index.ts

```diff
@@ -94,6 +96,45 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
         this.events.once("err", errorHandler);
     }
 
+    /**
+     * Builds out all the sub-environments for the given components. Components are saved into the calling interpreter
+     * instance. This function will mutate the state of the calling interpreter.
+     * @param componentMap Map of all components to be assigned to this interpreter
+     * @param manifest
+     * @param formatPathFn
+     * @param parseFn
+     */
+    public static async withSubEnvsFromComponents(
+        componentMap: Map<string, ComponentDefinition>,
+        formatPathFn: (comp: ComponentDefinition) => void,
```

Comment:
Is this passed in just to make testing easier?  If it never changes I'd rather change it to a helper function in this file and mock it in testing.

Reply:
There were a couple things:
1. Since it relies on the root value from the options and I was previously not passing it in it made sense to use a closure to encapsulate that value.
2. I don't think this is a function that the Interpreter needs to be intimate with. Ultimately I wanted to separate the concerns of reading, lexing, and parsing away from the actual interpretation of that data. 

I'd be interested in moving it into a helper function but I'd like to maintain that separation of responsibility.

Reply:
I ended up pulling this out and having the rewrite happen before the `Interpreter` is built.